### PR TITLE
migration: update cometbft tx size for t78 migration

### PR DESCRIPTION
**Pull request by conorsch**
## Describe your changes

We've already updated the CometBFT config template, but running nodes on the current testnet will still be using the prior value, generated at time of join. Let's update the value as part of `pd migrate`, logging a warning if we fail to do so.

## Issue ticket number and link
Refs #4594, #4582.

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

  > relevant to consensus-breaking logic and how it's handled via migrations

### CODE REVIEW

**Comment by conorsch** [comment]
Encountered some errors while testing this. In short, the TendermintConfig struct is stricter about type validation than CometBFT is. For example, `seed_mode = "false"` is accepted by CometBFT, but the struct demands `seed_mode = false`. Worse, an indexer node will fail, due to the `psql` indexer not being supported by tendermint-rs (#3188). Let's log warnings in those instances, rather than allowing the error to percolate up to bailing out on the top-level `pd migrate` action. If that still proves problematic, we can fall back to asking node operators to make the change manually.

File: crates/bin/pd/src/migrate/testnet78.rs [hunk 1]
```diff
@@ -520,8 +524,47 @@ fn truncate(s: &str, max_bytes: usize) -> &str {
     &s[..closest]
 }
 
-mod tests {
+/// Edit the node's CometBFT config file to set mempool.max_tx_bytes to the required value.
+/// This value will affect consensus, but the config setting is specified for CometBFT
```
Comment (dynst) [inline crates/bin/pd/src/migrate/testnet78.rs:528]:
```suggestion
/// Edit the node's CometBFT config file to set mempool.max_txs_bytes to the required value.
/// This value won't affect consensus, but the config setting is specified for CometBFT
```

Comment (dynst) [inline crates/bin/pd/src/migrate/testnet78.rs:528]:
```suggestion
/// Edit the node's CometBFT config file to set mempool.max_txs_bytes to the required value.
/// This value won't affect consensus, but the config setting is specified for CometBFT
```

**Comment by conorsch** [comment]
I'm happy with these changes: the cometbft munging will bail out before overwriting if it cannot interpret the on-disk cometbft file, logging a warning. That works well enough with the variety of munged configs I tested against.

File: crates/bin/pd/src/migrate/testnet78.rs [hunk 2]
```diff
@@ -536,8 +540,34 @@ fn truncate(s: &str, max_bytes: usize) -> &str {
...
+#[instrument]
+pub(crate) fn update_cometbft_mempool_settings(cometbft_home: PathBuf) -> anyhow::Result<()> {
+    let cometbft_config_path = cometbft_home.join("config").join("config.toml");
+    tracing::debug!(cometbft_config_path = %cometbft_config_path.display(), "opening cometbft config file");
+    let mut cometbft_config = TendermintConfig::load_toml_file(&cometbft_config_path)
+        .context("failed to load pre-migration cometbft config file")?;
+    // We expect the pre-migration value to be '1073741824', assuming it's unedited from `pd testnet join`.
+    // let pre_migration_value = cometbft_config.mempool.max_txs_bytes;
+    // The new value was updated in GH4594 and represents ~10MB or (2^20 * 10).
+    let post_migration_value = 10485760;
```
Comment (erwanor) [inline crates/bin/pd/src/migrate/testnet78.rs:555]: There are two `max_tx_bytes` parameters that we want to change:
- the first one is a `mempool` config parameter that controls the maximum size of a transaction to be included in the mempool/gossiped to other nodes 
- the second is a consensus parameter that controls the maximum size of block data (including headers)

We want to change both values:
- the mempool value should be set to something around 30KB (x-ref #4629)
- the consensus value should be set to 1MIB (x-ref #block size #4597)

**Review by erwanor — CHANGES_REQUESTED** [review]

File: crates/bin/pd/src/migrate/testnet78.rs (hunk 2 above)
Comment (dynst) [reply crates/bin/pd/src/migrate/testnet78.rs:555]: There are two different mempool configuration parameters that are easily mixed up.

- `max_txs_bytes` is what this PR is about
- `max_tx_bytes` is what #4629 is about, though it'd make sense to add it to this PR, too, after #4632 gets merged.

Comment (conorsch) [reply crates/bin/pd/src/migrate/testnet78.rs:555]: The `max_tx_bytes` vs `max_txs_bytes` distinction is certainly ripe for confusion, but if we're changing two values in the CometBFT config for Testnet 78, then we should update both values as part of the migration flow. I'll make a change to this PR to accommodate.

**Comment by conorsch** [comment]
This is ready for review again. After migration, the cometbft config's mempool block should look like:

```
[mempool]
recheck = true
broadcast = true
wal_dir = ""
size = 5000
max_txs_bytes = 10485760
cache_size = 10000
keep-invalid-txs-in-cache = false
max_tx_bytes = 30720
max_batch_bytes = 0
```

I made sure to test with a customized config that set `persistent_peers`, and those changes were preserved post-migration.

**Pull request merged**